File: Presentations/WG2_11/GlassBR_Code/fromGOOL/cpp/OutputFormat.cpp
#include "OutputFormat.hpp"

#include "InputParameters.hpp"

#include <algorithm>
#include <iostream>
#include <fstream>
#include <iterator>
#include <string>
#include <math.h>
#include <sstream>
#include <vector>

using namespace GlassBR;
using std::string;
using std::vector;
using std::ifstream;
using std::ofstream;

void GlassBR::display_output(string filename, double q, double j, double q_hat_tol, double pb, double lr, double nfl, bool is_safe1, bool is_safe2, InputParameters &inparams) {
    ofstream outfile;
    outfile.open(filename, std::fstream::out | std::fstream::app);
    outfile << "a           ";
    outfile << inparams.a << std::endl;
    outfile << "b           ";
    outfile << inparams.b << std::endl;
    outfile << "t           ";
    outfile << inparams.t << std::endl;
    outfile << "w           ";
    outfile << inparams.w << std::endl;
    outfile << "tnt         ";
    outfile << inparams.tnt << std::endl;
    outfile << "sdx         ";
    outfile << inparams.sdx << std::endl;
    outfile << "sdy         ";
    outfile << inparams.sdy << std::endl;
    outfile << "sdz         ";
    outfile << inparams.sdz << std::endl;
    outfile << "pbtol       ";
    outfile << inparams.pbtol << std::endl;
    outfile << "asprat      ";
    outfile << inparams.asprat << std::endl;
    outfile << "sd          ";
    outfile << inparams.sd << std::endl;
    outfile << "h           ";
    outfile << inparams.h << std::endl;
    outfile << "gtf         ";
    outfile << inparams.gtf << std::endl;
    outfile << "ldf         ";
    outfile << inparams.ldf << std::endl;
    outfile << "wtnt        ";
    outfile << inparams.wtnt << std::endl;
    outfile << "E           ";
    outfile << inparams.E << std::endl;
    outfile << "td          ";
    outfile << inparams.td << std::endl;
    outfile << "m           ";
    outfile << inparams.m << std::endl;
    outfile << "k           ";
    outfile << inparams.k << std::endl;
    outfile << "lsf         ";
    outfile << inparams.lsf << std::endl;
    outfile << "gt          ";
    outfile << inparams.gt << std::endl;
    outfile << "Demand (q)                      ";
    outfile << q << std::endl;
    outfile << "Stress Distr. Factor (j)        ";
    outfile << j << std::endl;
    outfile << "Tolerable Pressure (q_hat_tol)  ";
    outfile << q_hat_tol << std::endl;
    outfile << "Prob. of Breakage (pb)          ";
    outfile << pb << std::endl;
    outfile << "Capacity (lr)                   ";
    outfile << lr << std::endl;
    outfile << "Non-Factored Load (nfl)         ";
    outfile << nfl << std::endl;
    outfile << "Safety Req. 1 (is_safe1)        ";
    outfile << is_safe1 << std::endl;
    outfile << "Safety Req. 2 (is_safe2)        ";
    outfile << is_safe2 << std::endl;
    if (is_safe1 && is_safe2) {
        outfile << "For the given input parameters, the glass is considered safe." << std::endl;
    }
    else {
        outfile << "For the given input parameters, the glass is NOT considered safe." << std::endl;
    }
    outfile.close();
}

File: Presentations/WG2_11/GlassBR_Code/fromGOOL/python/OutputFormat.py
from __future__ import print_function
import sys
import math
import InputParameters


def display_output(filename, q, j, q_hat_tol, pb, lr, nfl, is_safe1, is_safe2, inparams):
    outfile = open(filename, "w")
    print("a           ", end='', file=outfile);
    print(inparams.a, file=outfile);
    print("b           ", end='', file=outfile);
    print(inparams.b, file=outfile);
    print("t           ", end='', file=outfile);
    print(inparams.t, file=outfile);
    print("w           ", end='', file=outfile);
    print(inparams.w, file=outfile);
    print("tnt         ", end='', file=outfile);
    print(inparams.tnt, file=outfile);
    print("sdx         ", end='', file=outfile);
    print(inparams.sdx, file=outfile);
    print("sdy         ", end='', file=outfile);
    print(inparams.sdy, file=outfile);
    print("sdz         ", end='', file=outfile);
    print(inparams.sdz, file=outfile);
    print("pbtol       ", end='', file=outfile);
    print(inparams.pbtol, file=outfile);
    print("asprat      ", end='', file=outfile);
    print(inparams.asprat, file=outfile);
    print("sd          ", end='', file=outfile);
    print(inparams.sd, file=outfile);
    print("h           ", end='', file=outfile);
    print(inparams.h, file=outfile);
    print("gtf         ", end='', file=outfile);
    print(inparams.gtf, file=outfile);
    print("ldf         ", end='', file=outfile);
    print(inparams.ldf, file=outfile);
    print("wtnt        ", end='', file=outfile);
    print(inparams.wtnt, file=outfile);
    print("E           ", end='', file=outfile);
    print(inparams.E, file=outfile);
    print("td          ", end='', file=outfile);
    print(inparams.td, file=outfile);
    print("m           ", end='', file=outfile);
    print(inparams.m, file=outfile);
    print("k           ", end='', file=outfile);
    print(inparams.k, file=outfile);
    print("lsf         ", end='', file=outfile);
    print(inparams.lsf, file=outfile);
    print("gt          ", end='', file=outfile);
    print(inparams.gt, file=outfile);
    print("Demand (q)                      ", end='', file=outfile);
    print(q, file=outfile);
    print("Stress Distr. Factor (j)        ", end='', file=outfile);
    print(j, file=outfile);
    print("Tolerable Pressure (q_hat_tol)  ", end='', file=outfile);
    print(q_hat_tol, file=outfile);
    print("Prob. of Breakage (pb)          ", end='', file=outfile);
    print(pb, file=outfile);
    print("Capacity (lr)                   ", end='', file=outfile);
    print(lr, file=outfile);
    print("Non-Factored Load (nfl)         ", end='', file=outfile);
    print(nfl, file=outfile);
    print("Safety Req. 1 (is_safe1)        ", end='', file=outfile);
    print(is_safe1, file=outfile);
    print("Safety Req. 2 (is_safe2)        ", end='', file=outfile);
    print(is_safe2, file=outfile);
    if (is_safe1 and is_safe2) :
        print("For the given input parameters, the glass is considered safe.", file=outfile);
    else :
        print("For the given input parameters, the glass is NOT considered safe.", file=outfile);
    outfile.close()


